### ISSUE TITLE
Features and Fixes

### ISSUE THREAD

**Issue by DRAGONTOS** [DRAGONTOS/stc#2]
# Features and Fixes

**Pls** if someone has a reply to a specific topic then use the corresponding **($NUMBER)**.
You can still give an better way of doing one of the **fixed** topics if relevant cus it could always be done in a better way :3  

### Steamcmd timeout. (1) 
wanna fix and have it rerun if it detects that steamcmd had a timeout and rerun
the timed out ids for 3x until it's fixed or if not it'll cout the ones and save it for a temp file to be run again using the --resume flag.

### Collections with one mod. (2) FIXED
So rn if you try to download a collection with one mod it will skip it and quit this is cus of the R"( \ )" string in the system function i could fix it by detecting the line count in the ids.txt file.

### Steamcmd dependency. (3) FIXED
So rn the steamcmd is locked to '$HOME/Steam/steamcmd.sh' which is the recommended way just not great if you don't have this or have it installed through the system's package manager. so i might want to make a config for this so that you can change it 

### Fix the help message. (4) FIXED
The help msg doesn't actually make sense rn, and it doesn't show the correct help msg for the given args and flags. (eg: -c & -m)

### Rename mods to their actual names. (5) FIXED
I want to rename the mods of a collection or in general to their actual names.

### Show downloaded mods 
Basically show the downloaded mods about the progress thingie.

#### Pull requests are always open if you want to make a draft on one of the topics, just say that you want to be added to the contributors or not.

**Comment by DRAGONTOS** [comment]
Fixed (2) in commit: 6004794ac970138e0a3513bc39ae7aff8c162a5b

**Comment by DRAGONTOS** [comment]
A debug option might be cool too.

**Comment by DRAGONTOS** [comment]
Fixed (4) in commit: ad2d685360bb2655023ad4960d2105661cb0be22

**Comment by DRAGONTOS** [comment]
Found a bug where it gives the error "terminate called after throwing an instance of 'char const*" when there is no network connection.

**Comment by DRAGONTOS** [comment]
Made the ground work for (1) in 1bb8e7ee49c2f638beb7165a51be1c35eaa820d7

**Comment by DRAGONTOS** [comment]
(3) might not ever be solved just have steamcmd installed in $HOME/Steam.

**Comment by DRAGONTOS** [comment]
-m seems to be broken

**Comment by DRAGONTOS** [comment]
could be my internet

**Comment by DRAGONTOS** [comment]
fixed (3) in commit: 1e8d348
